waftools/generators/headers.py

```diff
@@ -10,6 +10,15 @@ def __write_config_h__(ctx):
     __cp_to_variant__(ctx, ctx.options.variant, 'config.h')
     ctx.end_msg("config.h", "PINK")
 
+def __add_swift_defines__(ctx):
+    if ctx.dependency_satisfied("swift"):
```

Comment:
Ditto regarding early exit, since all of the code is under the if right now :)
```python
if not ctx.dependency_satisfied("swift"):
    return

...
```